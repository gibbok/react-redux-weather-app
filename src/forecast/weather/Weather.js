// @flow
import * as React from 'react'
import IconWind from '../../shared/icon/IconWind'
import moment from 'moment'
import IconWeather from '../../shared/icon/IconWeather'
import { withStyles, Typography } from 'material-ui'

/* eslint-disable no-undef */
type PropsType = {
  +weather:{
    +name: string,
    +country: string,
    +icon: number,
    +temperature: number,
    +temperatureMin: number,
    +temperatureMax: number,
    +weatherMain: number,
    +weatherIcon: number,
    +updatedTime:number,
    +windDegree: number,
    +windSpeed: number,
    +visibility: number,
    +humidity: number,
    +sunrise: number,
    +sunset: number,
    +cloudiness: number,
    +pressure: number,
    +pressureUnit: 'hPa'
  },
  +classes: Object
}
/* eslint-enable no-undef */

const styles = theme => {
  const spaceUnit = theme.spacing.unit
  return ({
    root: {
      display: 'flex',
      width: 900,
      marginBottom: spaceUnit * 3,
      flexDirection: 'column'
    },
    title: {
      display: 'flex',
      justifyContent: 'center'
    },
    temperature: {
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center'
    },
    temperatureContent: {
      display: 'flex',
      flexDirection: 'row',
      alignItems: 'center',
      justifyContent: 'space-between',
      width: 200
    },
    temperatureMinMax: {
      display: 'flex',
      flexDirection: 'column',
      width: 54
    },
    temperatureMin: {
      display: 'flex',
      justifyContent: 'space-between'
    },
    temperatureMax: {
      display: 'flex',
      justifyContent: 'space-between'
    },
    description: {
      display: 'flex',
      justifyContent: 'center'
    },
    update: {
      display: 'flex',
      justifyContent: 'center',
      marginTop: spaceUnit * 0.5,
      marginBottom: spaceUnit * 2.25
    },
    infos: {
      display: 'flex',
      justifyContent: 'center'
    },
    infosContent: {
      display: 'flex',
      flexDirection: 'column',
      width: 500
    },
    infosLine1: {
      display: 'flex',
      justifyContent: 'space-between',
      marginBottom: spaceUnit * 0.75
    },
    infosLine2: {
      display: 'flex',
      justifyContent: 'space-between'
    }
  })
}

const Weather:React.StatelessFunctionalComponent<any> = ({ weather: {
  name,
  country,
  icon,
  temperature,
  temperatureMin,
  temperatureMax,
  weatherMain,
  weatherIcon,
  updatedTime,
  windDegree,
  windSpeed,
  visibility,
  humidity,
  sunrise,
  sunset,
  cloudiness,
  pressure,
  pressureUnit
 },
 classes
}:PropsType):React.Element<any> => {
  const temperatureRounded = Math.round(temperature)
  const updatedTimeFormat = moment.unix(updatedTime).format('h:mm A')
  const sunriseFormat = moment.unix(sunrise).format('h:mm A')
  const sunsetFormat = moment.unix(sunset).format('h:mm A')
  const windSpeedFormat = Math.round(windSpeed)
  return (
    <div className={classes.root}>
      <div className={classes.title}>
        <Typography type='headline'>
          {name}, {country}
        </Typography>
      </div>
      <div className={classes.temperature}>
        <div className={classes.temperatureContent}>
          <div>
            <Typography type='display2'>
              <IconWeather code={icon} />
            </Typography>
          </div>
          <div>
            <Typography type='display3'>
              {temperatureRounded}&#176;
          </Typography>
          </div>
          <div className={classes.temperatureMinMax}>
            <div className={classes.temperatureMax}>
              <div><Typography type='body2'>Min</Typography></div>
              <div><Typography type='body2'>{temperatureMin} &#176;</Typography></div>
            </div>
            <div className={classes.temperatureMin}>
              <div><Typography type='body2'>Max</Typography></div>
              <div><Typography type='body2'>{temperatureMax} &#176;</Typography></div>
            </div>
          </div>
        </div>
      </div>
      <div className={classes.description}>
        <Typography type='title'>
          {weatherMain}
        </Typography>
      </div>
      <div className={classes.update}>
        <Typography type='body1'>
        Updated as of {updatedTimeFormat}
        </Typography>
      </div>
      <div className={classes.infos}>
        <div className={classes.infosContent}>
          <div className={classes.infosLine1}>
            <div>
              <Typography type='body1'>Wind <IconWind degree={windDegree} /> {windSpeedFormat} mps</Typography>
            </div>
            <div>
              <Typography type='body1'>Visibility {visibility} meter</Typography>
            </div>
            <div>
              <Typography type='body1'>Humidity {humidity} &#37;</Typography>
            </div>
          </div>
          <div className={classes.infosLine2}>
            <div>
              <Typography type='body1'>Sunrise {sunriseFormat}</Typography>
            </div>
            <div>
              <Typography type='body1'>Sunset {sunsetFormat}</Typography>
            </div>
            <div>
              <Typography type='body1'>Cloudiness {cloudiness} &#37;</Typography>
            </div>
            <div>
              <Typography type='body1'>Pressure {pressure} {pressureUnit}</Typography>
            </div>
          </div>
        </div>
      </div>
    </div>
  )
}

export default withStyles(styles)(Weather)
