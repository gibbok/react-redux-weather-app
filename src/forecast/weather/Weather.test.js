// @flow
import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Weather from './Weather'

describe('<Weather />', () => {
  it('should render', () => {
    const weather = {
      name: 'New York',
      country: 'US',
      icon: 800,
      temperature: 15.14,
      temperatureMax: 20,
      temperatureMin: 10,
      weatherMain: 'Clear',
      updatedTime: 1505196639,
      weatherIcon: 200,
      windDegree: 306.501,
      windSpeed: 1.29,
      visibility: 16093,
      humidity: 68,
      sunrise: 1505212456,
      sunset: 1505257754,
      cloudiness: 1,
      pressure: 1020,
      pressureUnit: 'hPa'
    }
    const wrapper = shallow(
      <Weather weather={weather} />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
