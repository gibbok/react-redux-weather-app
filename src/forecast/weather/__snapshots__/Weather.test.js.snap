// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Weather /> should render 1`] = `
<div
  className="Weather-root-1"
>
  <div
    className="Weather-title-2"
  >
    <withStyles(Typography)
      type="headline"
    >
      New York
      , 
      US
    </withStyles(Typography)>
  </div>
  <div
    className="Weather-temperature-3"
  >
    <div
      className="Weather-temperatureContent-4"
    >
      <div>
        <withStyles(Typography)
          type="display1"
        >
          <IconWeather
            code={800}
          />
        </withStyles(Typography)>
      </div>
      <div>
        <withStyles(Typography)
          type="display3"
        >
          15
          °
        </withStyles(Typography)>
      </div>
      <div
        className="Weather-temperatureMinMax-5"
      >
        <withStyles(Typography)
          type="subheading"
        >
          Min 
          10
           °
        </withStyles(Typography)>
        <withStyles(Typography)
          type="subheading"
        >
          Max 
          20
           °
        </withStyles(Typography)>
      </div>
    </div>
  </div>
  <div
    className="Weather-description-6"
  >
    <withStyles(Typography)
      type="title"
    >
      Clear
    </withStyles(Typography)>
  </div>
  <div
    className="Weather-update-7"
  >
    <withStyles(Typography)
      type="body1"
    >
      Updated as of 
      8:10 AM
    </withStyles(Typography)>
  </div>
  <div
    className="Weather-infos-8"
  >
    <div
      className="Weather-infosContent-9"
    >
      <div
        className="Weather-infosLine1-10"
      >
        <div>
          <withStyles(Typography)
            type="body1"
          >
            Wind 
            <IconWind
              degree={306.501}
            />
             
            1
             mps
          </withStyles(Typography)>
        </div>
        <div>
          <withStyles(Typography)
            type="body1"
          >
            Visibility 
            16093
             meter
          </withStyles(Typography)>
        </div>
        <div>
          <withStyles(Typography)
            type="body1"
          >
            Humidity 
            68
             %
          </withStyles(Typography)>
        </div>
      </div>
      <div
        className="Weather-infosLine2-11"
      >
        <div>
          <withStyles(Typography)
            type="body1"
          >
            Sunrise 
            12:34 PM
          </withStyles(Typography)>
        </div>
        <div>
          <withStyles(Typography)
            type="body1"
          >
            Sunset 
            1:09 AM
          </withStyles(Typography)>
        </div>
        <div>
          <withStyles(Typography)
            type="body1"
          >
            Cloudiness 
            1
             %
          </withStyles(Typography)>
        </div>
        <div>
          <withStyles(Typography)
            type="body1"
          >
            Pressure 
            1020
             
            hPa
          </withStyles(Typography)>
        </div>
      </div>
    </div>
  </div>
</div>
`;
